### DISCUSSION TITLE
Question about Morphometric calculator.

### DISCUSSION THREAD

**Discussion by Tao7410** [UMEP-dev/UMEP#490] [Q&A]
Hi
As a test, I use a grid(100m*100m) with a square building inside and I have tried to set the search throughout the grid extent.I would like to ask 2 questions: 
1. Why is the pai of each wind direction different, even equal to 0?
2. When finding the morphometric indices (pai, fai, zH, zHmax, zHstd, zd, z0 i wai) of each wind direction, how to generate the istropic.txt file?

thank you very much for the help
![image](https://user-images.githubusercontent.com/121487914/209670708-240be942-9b01-4168-b497-8f312c275db2.png)

**Comment by biglimp** [comment]
1. The concept of the calculation can be found [here](https://umep-docs.readthedocs.io/en/latest/pre-processor/Urban%20Morphology%20Morphometric%20Calculator%20(Grid).html). To keep consistency in a raster-based environment, we always calculate morphology in a north-south direction and instead rotating the grid using the scipy package in Python. I suspect that this could be the reason why some directions is a bit strange. Hopefully, the aggregated pai is correct.
2. The isotpophic file, you just aggregate all directions into one value (e.g. sum of column pai / number of wind directions).

Remember that these parameters should be used in the local scale and that 100 meter grid is closer to the micro scale and therefore, you might end up with strange values.
Hope this helps

**Reply by Tao7410** [reply]
Thank you for your help!
And one more question,do Millward-Hopkins method use Af*(unsheltered frontal area of roughness elements) or it's also use Af?

**Reply by biglimp** [reply]
I am not familiar with this. Please enlighten me.

**Reply by Tao7410** [reply]
In Millward-Hopkins et al. paper,Af is the sum of all the frontal areas and Af* is the frontal areas considering the mutual sheltering from the surrounding buildings.
![image](https://user-images.githubusercontent.com/121487914/215722507-45cf6cc4-1807-4fc8-a9a0-ac035993395b.png)
![image](https://user-images.githubusercontent.com/121487914/215722532-1632a0dc-44d7-47eb-bfee-a74f6bc9eba0.png)
![image](https://user-images.githubusercontent.com/121487914/215722562-311548cb-bfd2-44a9-865a-4cf7f063bbc4.png)
